vic/drivers/classic/src/alloc_veg_hist.c

```diff
@@ -38,29 +38,27 @@ alloc_veg_hist(int                nrecs,
 {
     int i, j;
 
-    (*veg_hist) = (veg_hist_struct **) calloc(nrecs, sizeof(veg_hist_struct *));
+    (*veg_hist) = calloc(nrecs, sizeof(*(*veg_hist)));
     if ((*veg_hist) == NULL) {
         log_err("Memory allocation error in alloc_veg_hist().");
     }
 
     for (i = 0; i < nrecs; i++) {
-        (*veg_hist)[i] = (veg_hist_struct *) calloc(nveg,
-                                                    sizeof(veg_hist_struct));
+        (*veg_hist)[i] = calloc(nveg, sizeof(*(*veg_hist)[i]));
```

Comment:
Note that a compiler would be unlikely to complain about either, since it will let you allocate any amount of memory that you want, even if it is the incorrect amount